Code/GraphMol/FileParsers/file_parsers_catch.cpp

```diff
@@ -431,6 +432,620 @@ M  END)CTAB";
+    std::unique_ptr<ROMol> mol(
+        MolBlockToMol(molblock2NoSBL, true, true, false));
+    REQUIRE(mol);
+    CHECK(MolToMolBlock(*mol) == expectedMolblock2NoSGroup1);
+  }
+  SECTION("molblock2 strictParsing true SBL group1 with >1 bond") {
+    std::unique_ptr<ROMol> mol;
+    REQUIRE_THROWS_AS(mol.reset(MolBlockToMol(molblock2BadSBL)),
+                      FileParseException);
+  }
+  SECTION("molblock2 strictParsing false SBL group1 with >1 bond") {
+    std::unique_ptr<ROMol> mol(
+        MolBlockToMol(molblock2BadSBL, true, true, false));
+    REQUIRE(mol);
+    CHECK(MolToMolBlock(*mol) == expectedMolblock2NoSGroup1);
+  }
+}
+
+TEST_CASE(
+    "do not throw but remove malformed V2000 SGroups when strictParsing is "
+    "false",
+    "[feature][sgroups]") {
+  std::string molblock = R"CTAB(
+  ChemDraw01072117362D
+
+ 16 16  0  0  0  0  0  0  0  0999 V2000
+   -1.7862    0.7219    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -1.7862   -0.1031    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -1.0717   -0.5156    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3572   -0.1031    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3572    0.7219    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -1.0717    1.1344    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.3572    2.3719    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
+    0.3572   -0.5156    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    1.0717   -0.1031    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
+    0.3572   -1.3406    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
+    1.0717   -1.7531    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    1.0717   -2.5781    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+    1.7862   -1.3406    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -1.4289    1.3406    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.7145    1.7531    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.7145    2.5781    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+ 11 12  1  0      
+ 11 13  1  0      
+ 10 11  1  0      
+  8  9  2  0      
+  8 10  1  0      
+  4  8  1  0      
+ 15 16  1  0      
+ 14 15  1  0      
+  6 14  1  0      
+  7 16  1  0      
+  1  2  2  0      
+  2  3  1  0      
+  3  4  2  0      
+  4  5  1  0      
+  5  6  2  0      
+  6  1  1  0      
+M  STY  1   1 SUP
+M  SLB  1   1   1
+M  SAL   1  6   8   9  10  11  12  13
+M  SBL   1  1   6
+M  SMT   1 COOiPr
+M  SBV   1   6   -0.7145    0.4125
+M  STY  1   2 SUP
+M  SLB  1   2   2
+M  SAL   2  3  14  15  16
+M  SBL   2  2   9  10
+M  SMT   2 (CH2)3
+M  SBV   2   9    0.3572   -0.2062
+M  SBV   2  10    0.3572   -0.2062
+M  END
+)CTAB";
+  SECTION("molblock strictParsing true") {
+    std::unique_ptr<ROMol> mol(MolBlockToMol(molblock));
+    REQUIRE(mol);
+    CHECK(getSubstanceGroups(*mol).size() == 2);
+  }
+  SECTION("molblock bad sgroup idx") {
+    std::string molblockBad = std::regex_replace(
```

Comment:
I don't see why you are using `std::regex` here (or elsewhere in this file). It looks like you are doing normal string manipulations, not regex work.

There's an actual risk here: using `std::regex` causes problems with the earliest g++ version with which it's possible to (barely) compile the RDKit - I think that's v4.8.x. That version of g++ has decent c++11 support in the compiler, but the library implementation, particularly the regex implementation is incomplete.
when not required causes

Reply:
OK, I wasn't aware of that.

Reply:
Did I miss some place where you were actually using regex functionality?

Reply:
No, you didn't. The dumbness is all on my side : )